realm/src/main/java/io/realm/internal/TableView.java

```diff
@@ -73,19 +73,24 @@
      * @param parent A table.
      * @param nativePtr pointer to table.
      */
-    protected TableView(Context context, Table parent, long nativePtr){
+    protected TableView(Context context, Table parent, long nativePtr) {
         this.context = context;
         this.parent = parent;
         this.nativePtr = nativePtr;
     }
 
+    protected TableView(Context context, Table parent, long nativePtr, TableQuery query) {
```

Comment:
I think I understand that. I am just wondering why the TableQuery doesn't hold that reference already, saving the complicated constructor.